### ISSUE TITLE
The vocab.txt file for norwegian is missing some tokens

### ISSUE THREAD

**Issue by Sleemanmunk** [mollerhoj/danish_bert#9]
I'm really excited about this model. Hoping to use it to help me write a sentiment model for a Norwegian company.
I'm getting a lot of UNK when I try to use your Norwegian model.
I looked at vocab.txt, and it seems to have only ## words.

**Comment by marasva** [comment]
Hi! 
Encountered the same problem, but seems like you can fix this by writing a regex function removing "##▁" from the vocab.txt-file. Regex example: newText = re.sub(r'##▁', "", text). (Notice the bolded underline)

I noticed that the danish text file did not have the "##_", so figured it was worth just trying to remove them.

**Comment by Sleemanmunk** [comment]
Does it work well when you use it?

**Comment by marasva** [comment]
The tokenizer works well, but I cannot get the models to work properly no. They predict only one class, and I have tried it for both the Danish and Norwegian BERT model with both being equally bad. I followed [this](https://www.kdnuggets.com/2020/02/intent-recognition-bert-keras-tensorflow.html) tutorial and managed to get it to work with the example, but not my own models. 

When the model only predict one class with the same probability each time someone suggested that it would maybe have a problem loading the BERT weights, which seems probable, but I do not know where to go further.

**Comment by Sleemanmunk** [comment]
When you followed the tutorial, did you get nordic bert to work with the example or English bert?If you've never gotten nordic bert to work, I think your solution didn't really work, and the authors need to fix their model.

**Comment by marasva** [comment]
Just the English BERT, that is why I also think there is something wrong with the Nordic BERT models. But the principle of that tutorial should be more or less the same? I managed to fix the tokenization problem with the Norwegian model, but nothing further.

**Comment by Sleemanmunk** [comment]
Well, it appears that you fixed it, i.e. it looks right now, but it could be that it only looks right and in fact it's still wrong, as the inability to make a successful model with it suggests.

**Comment by marasva** [comment]
Yes, because trying the Danish version should more or less work with a Norwegian dataset considering many similar words, but that did not work either.

**Comment by mollerhoj** [comment]
I guys. I'm am not exactly sure what issue you are having.

I'm quite busy these days, but will get back to you once I've had a look at the models.

I can confirm that the models should be able to replace the standard english model bert model, and that many users have succeeded in doing so.

**Comment by marasva** [comment]
That may be, but as mentioned I have managed to get a basic example working with the English BERT model, and the exact same procedure (just changing dataset) does not work with the Norwegian or Danish BERT model. We chose to not go forward with BERT due to these complications.

**Comment by christianbv** [comment]
I don't think you should replace both ## and _ at the same time. Appears as if ## should be before all words having _, whereas the words without _ should have no ## -> e.g. 

word = word.replace("#","")
word = word.replace("_","##")

**Comment by Sleemanmunk** [comment]
Nice suggestion! Let us know if it leads to good models!

**Comment by marasva** [comment]
As I have already explained previously the tokenization seemed to work as it should doing what I did, but I could, of course, be wrong. However, the "problem" as I see it (overlooking the Norwegian tokenization problems) is that the Danish model does not work like one should expect following the "standard" English BERT model since it does not have these tokenization problems. Of course, I could just be implementing it all wrong, but it still has all the same files as the "official" English BERT model and I do not have enough expertise to know why the models do not work.

**Comment by christianbv** [comment]
I forgot to mention that I used the Huggingface transformers library - it worked for me :)

**Comment by Sleemanmunk** [comment]
When you say that it worked, do you mean that you tried modeling with it
and got good results or just that it didn't crash?

**Comment by christianbv** [comment]
Yes meaning that the model doesn't crash. I have the same issue as the one mentioned earlier above; the model keeps predicting the same class for a given epoch even though the classes are balanced and the number of epochs are high..

**Comment by christinaknudsen** [comment]
I tried the Nordic models for text generation. The Danish and the Swedish models seem to work fine, but the Norwegian model only returns \<unk> tokens. I tried replacing ##_ in the vocab file, and now the tokenizer works, but the model still seems off (returning only \<unk>). (link to my github: https://github.com/christinaknudsen/chatbot_research/tree/master/transformers).

Did anyone find out what the problem might be?

**Comment by Sleemanmunk** [comment]
It's broken. Chances are good that it was broken when Jens Dahl Moller was
training it in the first place, which means that there's no recovery
without training again from scratch.
As far as I can tell, the only way to get it to work is to put pressure on
the creator to fix it. Please email him at jdm@botxo.co and tell him how
much having a functioning Norwegian
BERT model would mean to your research.

Sam

**Comment by akutuzov** [comment]
For reference of those who found this issue while searching for Norwegian BERT: 

you can find a fully functional Bert model trained on Norwegian texts at https://huggingface.co/ltgoslo/norbert

An academic paper describing this model will be published ion the nearest days.

**Comment by Sleemanmunk** [comment]
Great news! Thanks!

**Comment by akutuzov** [comment]
The paper about NorBERT is available here:

https://arxiv.org/abs/2104.06546
